### PR TITLE
Added makefiles and instructions to compile in OS X

### LINKED ISSUE #37: Build in *nix
Necessaey Makefiles etc to build and run eunit/ct tests in *nix(es) like dardwin, ubuntu, RHEL etc

### PR DESCRIPTION
**Pull request by acautin**
Fixes #37

### CODE REVIEW

File: c_src/Makefile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+T = test
+ODPI_LIB_DIR = $(S)/odpi/lib
+
+OBJS = $(O)/dpi_nif.o $(O)/dpiContext_nif.o $(O)/dpiConn_nif.o $(O)/dpiStmt_nif.o $(O)/dpiData_nif.o $(O)/dpiQueryInfo_nif.o $(O)/dpiVar_nif.o $(O)/dpiObjectType_nif.o
+TARGETS = $(O)/dpi_nif.so
+
+TEST_OBJS = test/odpi_test.o
+TEST_TARGETS = test/odpi_test
+
+BASECXXFLAGS = -ggdb -Wall -fPIC -shared -I$(S) -I$(ERL_INTERFACE_DIR)/include -I"$(OTP_ERTS_DIR)/include" -I"$(S)/odpi/include"
```
Comment (c-bik) [inline c_src/Makefile]: Please also split this into `INCLUDEDIRS`

File: c_src/Makefile [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+$(TARGETS): odpi $(ORANIF_SRCS)
+	gcc -o $@ $(CXXFLAGS) $(LINKDIRS) $(LINKFLAGS) $(ORANIF_SRCS)
+
+#$(TEST_TARGETS): odpi $(TEST_OBJS)
+
+priv:
+	$(MKDIR_P) $(O)
+
+odpi:
+	@if [ ! -d "c_src/odpi" ] ; then cd c_src && (git clone -b v3.0.0 --single-branch https://github.com/K2InformaticsGmbH/odpi) fi
```
Comment (c-bik) [inline c_src/Makefile]: Preferred multiline for readablity

**Comment by acautin** [comment]
@c-bik updated please verify.

**Comment by shamis** [comment]
not able to compile on centos
```
rebar3 compile
===> Verifying dependencies...
===> Compiling oranif
cd c_src/odpi && make
make[1]: Entering directory `/home/shamis/repo/tmp/oranif/c_src/odpi'
make[1]: Nothing to be done for `all'.
make[1]: Leaving directory `/home/shamis/repo/tmp/oranif/c_src/odpi'
gcc -o priv/dpi_nif.so -ggdb -Wall -fPIC -shared -Ic_src -I/usr/lib/erlang/lib/erl_interface-3.10.1/include -I"/usr/lib/erlang/erts-9.2/include" -I"c_src/odpi/include" -L/usr/lib/erlang/lib/erl_interface-3.10.1/lib -Lc_src/odpi/lib -lerl_interface -lei -lodpic c_src/dpi_nif.c c_src/dpiContext_nif.c c_src/dpiConn_nif.c c_src/dpiStmt_nif.c c_src/dpiData_nif.c c_src/dpiQueryInfo_nif.c c_src/dpiVar_nif.c c_src/dpiObjectType_nif.c
c_src/dpiContext_nif.c: In function â��dpiContext_create_nifâ��:
c_src/dpiContext_nif.c:21: warning: pointer targets in passing argument 3 of â��enif_get_uintâ�� differ in signedness
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:103: note: expected â��unsigned int *â�� but argument is of type â��int *â��
c_src/dpiContext_nif.c:23: warning: pointer targets in passing argument 3 of â��enif_get_uintâ�� differ in signedness
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:103: note: expected â��unsigned int *â�� but argument is of type â��int *â��
c_src/dpiContext_nif.c: In function â��dpiContext_getError_nifâ��:
c_src/dpiContext_nif.c:53: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiContext_res **â��
c_src/dpiContext_nif.c: In function â��dpiContext_destroy_nifâ��:
c_src/dpiContext_nif.c:67: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiContext_res **â��
c_src/dpiContext_nif.c: In function â��dpiContext_getClientVersion_nifâ��:
c_src/dpiContext_nif.c:81: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiContext_res **â��
c_src/dpiConn_nif.c: In function â��dpiConn_create_nifâ��:
c_src/dpiConn_nif.c:24: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiContext_res **â��
c_src/dpiConn_nif.c:34: warning: pointer targets in passing argument 2 of â��dpiConn_createâ�� differ in signedness
c_src/odpi/include/dpi.h:769: note: expected â��const char *â�� but argument is of type â��unsigned char *â��
c_src/dpiConn_nif.c:34: warning: pointer targets in passing argument 4 of â��dpiConn_createâ�� differ in signedness
c_src/odpi/include/dpi.h:769: note: expected â��const char *â�� but argument is of type â��unsigned char *â��
c_src/dpiConn_nif.c:34: warning: pointer targets in passing argument 6 of â��dpiConn_createâ�� differ in signedness
c_src/odpi/include/dpi.h:769: note: expected â��const char *â�� but argument is of type â��unsigned char *â��
c_src/dpiConn_nif.c: In function â��dpiConn_prepareStmt_nifâ��:
c_src/dpiConn_nif.c:57: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiConn_res **â��
c_src/dpiConn_nif.c:69: warning: implicit declaration of function â��strcmpâ��
c_src/dpiConn_nif.c:69: warning: pointer targets in passing argument 3 of â��dpiConn_prepareStmtâ�� differ in signedness
c_src/odpi/include/dpi.h:852: note: expected â��const char *â�� but argument is of type â��unsigned char *â��
c_src/dpiConn_nif.c:69: warning: pointer targets in passing argument 5 of â��dpiConn_prepareStmtâ�� differ in signedness
c_src/odpi/include/dpi.h:852: note: expected â��const char *â�� but argument is of type â��unsigned char *â��
c_src/dpiConn_nif.c: In function â��dpiConn_newVar_nifâ��:
c_src/dpiConn_nif.c:96: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiConn_res **â��
c_src/dpiConn_nif.c:112: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiObjectType_res **â��
c_src/dpiConn_nif.c:128: error: â��forâ�� loop initial declarations are only allowed in C99 mode
c_src/dpiConn_nif.c:128: note: use option -std=c99 or -std=gnu99 to compile your code
c_src/dpiConn_nif.c:91: warning: unused variable â��isArrayBufâ��
c_src/dpiConn_nif.c: In function â��dpiConn_commit_nifâ��:
c_src/dpiConn_nif.c:152: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiConn_res **â��
c_src/dpiConn_nif.c: In function â��dpiConn_rollback_nifâ��:
c_src/dpiConn_nif.c:165: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiConn_res **â��
c_src/dpiConn_nif.c: In function â��dpiConn_ping_nifâ��:
c_src/dpiConn_nif.c:179: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiConn_res **â��
c_src/dpiConn_nif.c: In function â��dpiConn_release_nifâ��:
c_src/dpiConn_nif.c:193: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiConn_res **â��
c_src/dpiConn_nif.c: In function â��dpiConn_close_nifâ��:
c_src/dpiConn_nif.c:209: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiConn_res **â��
c_src/dpiConn_nif.c:230: warning: pointer targets in passing argument 3 of â��dpiConn_closeâ�� differ in signedness
c_src/odpi/include/dpi.h:762: note: expected â��const char *â�� but argument is of type â��unsigned char *â��
c_src/dpiConn_nif.c: In function â��dpiConn_getServerVersion_nifâ��:
c_src/dpiConn_nif.c:245: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiConn_res **â��
c_src/dpiConn_nif.c:252: warning: passing argument 2 of â��dpiConn_getServerVersionâ�� from incompatible pointer type
c_src/odpi/include/dpi.h:818: note: expected â��const char **â�� but argument is of type â��char **â��
c_src/dpiConn_nif.c:252: warning: pointer targets in passing argument 3 of â��dpiConn_getServerVersionâ�� differ in signedness
c_src/odpi/include/dpi.h:818: note: expected â��uint32_t *â�� but argument is of type â��int *â��
c_src/dpiStmt_nif.c: In function â��dpiStmt_execute_nifâ��:
c_src/dpiStmt_nif.c:25: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiStmt_res **â��
c_src/dpiStmt_nif.c: In function â��dpiStmt_fetch_nifâ��:
c_src/dpiStmt_nif.c:59: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiStmt_res **â��
c_src/dpiStmt_nif.c: In function â��dpiStmt_getQueryValue_nifâ��:
c_src/dpiStmt_nif.c:87: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiStmt_res **â��
c_src/dpiStmt_nif.c: In function â��dpiStmt_getQueryInfo_nifâ��:
c_src/dpiStmt_nif.c:125: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiStmt_res **â��
c_src/dpiStmt_nif.c: In function â��dpiStmt_bindValueByPos_nifâ��:
c_src/dpiStmt_nif.c:149: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiStmt_res **â��
c_src/dpiStmt_nif.c:153: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiData_res **â��
c_src/dpiStmt_nif.c: In function â��dpiStmt_bindValueByName_nifâ��:
c_src/dpiStmt_nif.c:174: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiStmt_res **â��
c_src/dpiStmt_nif.c:178: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiData_res **â��
c_src/dpiStmt_nif.c:184: warning: pointer targets in passing argument 2 of â��dpiStmt_bindValueByNameâ�� differ in signedness
c_src/odpi/include/dpi.h:1577: note: expected â��const char *â�� but argument is of type â��unsigned char *â��
c_src/dpiStmt_nif.c: In function â��dpiStmt_bindByPos_nifâ��:
c_src/dpiStmt_nif.c:199: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiStmt_res **â��
c_src/dpiStmt_nif.c:203: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiVar_res **â��
c_src/dpiStmt_nif.c: In function â��dpiStmt_bindByName_nifâ��:
c_src/dpiStmt_nif.c:219: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiStmt_res **â��
c_src/dpiStmt_nif.c:223: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiVar_res **â��
c_src/dpiStmt_nif.c:226: warning: pointer targets in passing argument 2 of â��dpiStmt_bindByNameâ�� differ in signedness
c_src/odpi/include/dpi.h:1568: note: expected â��const char *â�� but argument is of type â��unsigned char *â��
c_src/dpiStmt_nif.c: In function â��dpiStmt_release_nifâ��:
c_src/dpiStmt_nif.c:238: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiStmt_res **â��
c_src/dpiStmt_nif.c:242: warning: statement with no effect
c_src/dpiStmt_nif.c: In function â��dpiStmt_define_nifâ��:
c_src/dpiStmt_nif.c:255: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiStmt_res **â��
c_src/dpiStmt_nif.c:259: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiVar_res **â��
c_src/dpiStmt_nif.c: In function â��dpiStmt_defineValue_nifâ��:
c_src/dpiStmt_nif.c:279: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiStmt_res **â��
c_src/dpiStmt_nif.c:289: warning: implicit declaration of function â��strcmpâ��
c_src/dpiData_nif.c: In function â��dpiData_setTimestamp_nifâ��:
c_src/dpiData_nif.c:53: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiDataPtr_res **â��
c_src/dpiData_nif.c:55: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiData_res **â��
c_src/dpiData_nif.c: In function â��dpiData_setIntervalDS_nifâ��:
c_src/dpiData_nif.c:96: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiDataPtr_res **â��
c_src/dpiData_nif.c:100: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiData_res **â��
c_src/dpiData_nif.c: In function â��dpiData_setIntervalYM_nifâ��:
c_src/dpiData_nif.c:134: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiDataPtr_res **â��
c_src/dpiData_nif.c:138: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiData_res **â��
c_src/dpiData_nif.c: In function â��dpiData_setInt64_nifâ��:
c_src/dpiData_nif.c:165: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiDataPtr_res **â��
c_src/dpiData_nif.c:169: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiData_res **â��
c_src/dpiData_nif.c: In function â��dpiData_setBytes_nifâ��:
c_src/dpiData_nif.c:192: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiDataPtr_res **â��
c_src/dpiData_nif.c:196: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiData_res **â��
c_src/dpiData_nif.c:209: warning: pointer targets in passing argument 2 of â��dpiData_setBytesâ�� differ in signedness
c_src/odpi/include/dpi.h:955: note: expected â��char *â�� but argument is of type â��unsigned char *â��
c_src/dpiData_nif.c: In function â��dpiData_setIsNull_nifâ��:
c_src/dpiData_nif.c:222: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiDataPtr_res **â��
c_src/dpiData_nif.c:226: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiData_res **â��
c_src/dpiData_nif.c:237: warning: implicit declaration of function â��strcmpâ��
c_src/dpiData_nif.c: In function â��dpiData_get_nifâ��:
c_src/dpiData_nif.c:248: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiDataPtr_res **â��
c_src/dpiData_nif.c:275: warning: implicit declaration of function â��memcpyâ��
c_src/dpiData_nif.c:275: warning: incompatible implicit declaration of built-in function â��memcpyâ��
c_src/dpiData_nif.c: In function â��dpiData_getInt64_nifâ��:
c_src/dpiData_nif.c:362: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiDataPtr_res **â��
c_src/dpiData_nif.c:366: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiData_res **â��
c_src/dpiData_nif.c: In function â��dpiData_getBytes_nifâ��:
c_src/dpiData_nif.c:388: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiDataPtr_res **â��
c_src/dpiData_nif.c:390: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiData_res **â��
c_src/dpiData_nif.c:400: warning: incompatible implicit declaration of built-in function â��memcpyâ��
c_src/dpiData_nif.c: In function â��dpiData_release_nifâ��:
c_src/dpiData_nif.c:414: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiData_res **â��
c_src/dpiData_nif.c:418: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiDataPtr_res **â��
c_src/dpiQueryInfo_nif.c: In function â��dpiQueryInfo_get_nifâ��:
c_src/dpiQueryInfo_nif.c:22: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiQueryInfo_res **â��
c_src/dpiQueryInfo_nif.c: In function â��dpiQueryInfo_delete_nifâ��:
c_src/dpiQueryInfo_nif.c:86: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiQueryInfo_res **â��
c_src/dpiVar_nif.c: In function â��dpiVar_setNumElementsInArray_nifâ��:
c_src/dpiVar_nif.c:19: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiVar_res **â��
c_src/dpiVar_nif.c:21: warning: pointer targets in passing argument 3 of â��enif_get_intâ�� differ in signedness
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:43: note: expected â��int *â�� but argument is of type â��uint32_t *â��
c_src/dpiVar_nif.c: In function â��dpiVar_setFromBytes_nifâ��:
c_src/dpiVar_nif.c:38: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiVar_res **â��
c_src/dpiVar_nif.c:40: warning: pointer targets in passing argument 3 of â��enif_get_intâ�� differ in signedness
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:43: note: expected â��int *â�� but argument is of type â��uint32_t *â��
c_src/dpiVar_nif.c:45: warning: pointer targets in passing argument 3 of â��dpiVar_setFromBytesâ�� differ in signedness
c_src/odpi/include/dpi.h:1729: note: expected â��const char *â�� but argument is of type â��unsigned char *â��
c_src/dpiVar_nif.c: In function â��dpiVar_release_nifâ��:
c_src/dpiVar_nif.c:57: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiVar_res **â��
c_src/dpiObjectType_nif.c: In function â��dpiObjectType_release_nifâ��:
c_src/dpiObjectType_nif.c:19: warning: passing argument 4 of â��enif_get_resourceâ�� from incompatible pointer type
/usr/lib/erlang/erts-9.2/include/erl_nif_api_funcs.h:114: note: expected â��void **â�� but argument is of type â��struct dpiObjectType_res **â��
make: *** [priv/dpi_nif.so] Error 1
===> Hook for compile failed!
```

**Comment by acautin** [comment]
@shamis, thanks will give it a look

**Comment by acautin** [comment]
@c-bik do we need compatibility with CentOS 6.9 and old gcc ? `gcc version 4.4.7 20120313 (Red Hat 4.4.7-18) (GCC)` I added a flag to compile against the c11 standard so `gcc 4.7+` will be required. What do you think ?

**Comment by c-bik** [comment]
> @c-bik do we need compatibility with CentOS 6.9 and old gcc ? `gcc version 4.4.7 20120313 (Red Hat 4.4.7-18) (GCC)` I added a flag to compile against the c11 standard so `gcc 4.7+` will be required. What do you think ?

Lets got for the latest Compiler support. `c11` flag enough as fix!

**Pull request merged**